pkg/tsdb/zipkin/zipkin.go

```diff
@@ -0,0 +1,81 @@
+package zipkin
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/grafana/grafana-plugin-sdk-go/backend"
+	"github.com/grafana/grafana-plugin-sdk-go/backend/datasource"
+	"github.com/grafana/grafana-plugin-sdk-go/backend/instancemgmt"
+
+	"github.com/grafana/grafana/pkg/infra/httpclient"
+)
+
+type Service struct {
+	im instancemgmt.InstanceManager
+}
+
+func ProvideService(httpClientProvider httpclient.Provider) *Service {
+	return &Service{
+		im: datasource.NewInstanceManager(newInstanceSettings(httpClientProvider)),
+	}
+}
+
+type datasourceInfo struct {
+	ZipkinClient ZipkinClient
+}
+
+func newInstanceSettings(httpClientProvider httpclient.Provider) datasource.InstanceFactoryFunc {
+	return func(ctx context.Context, settings backend.DataSourceInstanceSettings) (instancemgmt.Instance, error) {
+		httpClientOptions, err := settings.HTTPClientOptions(ctx)
+		if err != nil {
+			return nil, backend.DownstreamError(fmt.Errorf("error reading settings: %w", err))
+		}
+
+		httpClient, err := httpClientProvider.New(httpClientOptions)
+		if err != nil {
+			return nil, fmt.Errorf("error creating http client: %w", err)
+		}
+
+		if settings.URL == "" {
+			return nil, backend.DownstreamError(fmt.Errorf("error reading settings: url is empty"))
```

Comment:
Nit: There's just a few places where errors are created but don't actually need formatting 😄 

```suggestion
			return nil, backend.DownstreamError(errors.New("error reading settings: url is empty"))
```

Reply:
👍 , fixed also in https://github.com/grafana/grafana/pull/96031/commits/61419f6e306b3119cceb6a5ffce953b7cd03fbda